quicfire_tools/inputs.py

```diff
@@ -621,28 +636,184 @@ def set_output_files(
+                    float,
+                    checks=Check(lambda s: len(s.unique()) == 1, element_wise=False),
+                ),
+                "x_location": Column(
+                    int,
+                    checks=Check(lambda s: len(s.unique()) == 1, element_wise=False),
+                ),
+                "y_location": Column(
+                    int,
+                    checks=Check(lambda s: len(s.unique()) == 1, element_wise=False),
+                ),
+            }
+        )
+        validated_df = validate(df)
+        if update is None:
+            self.windsensors.add_sensor(
+                wind_times=list(validated_df["wind_times"]),
+                wind_speeds=list(validated_df["wind_speeds"]),
+                wind_directions=list(validated_df["wind_directions"]),
+                sensor_height=list(validated_df["sensor_height"])[0],
+                x_location=list(validated_df["x_location"])[0],
+                y_location=list(validated_df["y_location"])[0],
+            )
+        else:
+            self.windsensors.update_sensor(
+                sensor_name=update,
+                wind_times=list(validated_df["wind_times"]),
+                wind_speeds=list(validated_df["wind_speeds"]),
+                wind_directions=list(validated_df["wind_directions"]),
+                sensor_height=list(validated_df["sensor_height"])[0],
+                x_location=list(validated_df["x_location"])[0],
+                y_location=list(validated_df["y_location"])[0],
+            )
+
     def _update_shared_attributes(self):
         self.gridlist.n = self.qu_simparams.nx
         self.gridlist.m = self.qu_simparams.ny
         self.gridlist.l = self.quic_fire.nz
         self.gridlist.dx = self.qu_simparams.dx
         self.gridlist.dy = self.qu_simparams.dy
-        if (
-            not self.sensor1.time_now
-            == self.quic_fire.time_now
-            == self.qu_simparams.wind_times[0]
-        ):
-            # TODO: How to handle conflicts
+        if not self.windsensors.time_now == self.quic_fire.time_now:
             print(
                 f"WARNING: fire start time must be the same for all input files.\n"
                 f"Times: \n"
                 f"\tQUIC_fire.inp: {self.quic_fire.time_now}\n"
                 f"\tQU_simparams.inp: {self.qu_simparams.wind_times[0]}\n"
-                f"\tsensor1.inp: {self.sensor1.time_now}\n"
+                f"\twindsensor.inp: {self.windsensors.time_now}\n"
                 f"Setting all values to {self.quic_fire.time_now}"
             )
-            self.sensor1.time_now = self.quic_fire.time_now
-            self.qu_simparams.wind_times[0] = self.quic_fire.time_now
+            self.windsensors.time_now = self.quic_fire.time_now
+        self.qu_simparams.wind_times = [
+            s + self.windsensors.time_now for s in self.windsensors.wind_times
+        ]
+        self.qu_metparams.num_sensors = len(self.windsensors.sensor_array)
```

Comment:
We don't need to address it now, but I really want to find a way to get rid of this.

Reply:
No more shared attribute issues with the windsensors themselves. The issue now is with qu_simparams and qu_metparams